Modules/RosterManager/Roster.lua

```diff
@@ -83,6 +87,53 @@ function Roster:Standings(GUID)
     end
 end
 
+function Roster:WeeklyGains(GUID, week)
+    if type(GUID) ~= "string" then
+        return self.weeklyGains or {}
+    end
+    if type(week) ~= "number" then
+        return self.weeklyGains[GUID] or {}
+    end
+    return self.weeklyGains[GUID][week] or 0
+end
+
+function Roster:UpdateStandings(GUID, value, timestamp)
+    local offset = (self.configuration._.weeklyReset == CONSTANTS.WEEKLY_RESET.EU) and weekOffsetEU or weekOffsetUS
+    local week = WeekNumber(timestamp, offset)
+    local hardCap = self.configuration._.hardCap
+    local isPositiveChange = (value > 0)
+    -- We do not remove points if they are over during newly introduced cap
+    if self.configuration.hasHardCap and (self:Standings(GUID) >  hardCap) and isPositiveChange then
+        return
+    end
+    self.standings[GUID] = self:Standings(GUID) + value
+    if isPositiveChange then
+        -- Weekly Cap
+        if self.configuration.hasWeeklyCap and timestamp then
+            self.weeklyGains[GUID][week] = self:WeeklyGains(GUID, week) + value
+            local overCap = self:WeeklyGains(GUID, week) - self.configuration._.weeklyCap
+            if overCap > 0 then
+                self.weeklyGains[GUID][week] = self:WeeklyGains(GUID, week) - overCap
+                self.standings[GUID] = self:Standings(GUID) - overCap
+            end
+        end
+        -- Hard Cap
+        if self.configuration.hasHardCap then
```

Comment:
I think you must check hard cap first.
Specifically if someone hits the hard cap it is still counted as their weekly gains.
Imagine weekly cap 30, current balance 30, hard cap 40

- reward 20 (sets weekly gains to 20, balance to 40)
- award item for 40 (sets balance to 0)
- reward 20 (sets weekly gains to 30, balance to 10)

Reply:
Raid.lua:106 covers that

Reply:
I don't see it... link?
And it shouldn't be covered in some other module.
This code is responsible for tracking weekly gains and it doesn't do that properly.
If I don't get DKP due to a hard cap, then it should not be counted for my weekly gains.

Reply:
Wrote raid meant roster
https://github.com/ClassicLootManager/ClassicLootManager/blob/feature-point-hard-weekly-cap/Modules/RosterManager/Roster.lua#L106
This covers hard cap
and yes if you hit hard cap i still count the weekly gain as you get those points. just your standings are not capped. also all the history entries will have those original values (so if you have cap of 400, and 350 standing, get 100 DKP, then you get 100 this week, 100 in history but your stangins change only to 400)

Reply:
Taking this to #50.